### ISSUE TITLE
Import support for Creo View files.

### ISSUE THREAD

**Issue by naikrovek** [assimp/assimp#1891]
Feature Request.  Creo View files can be converted to ASCII with a tool downloadable from PTC.  Creo View downloads come with example model files.  I can also supply example files, if required.

Please support the import of PTC Creo View files; PVZ, PVS, and OL.

**Comment by kimkulling** [comment]
Interesting idea, thanks for the feature request!

**Comment by PrinzDavid** [comment]
@naikrovek 
Could you tell me, where that tool is downloadable?
Thanks in advance!

**Comment by naikrovek** [comment]
you will need to have a PTC account to get access to it, I believe.  it's called "Creo View Files Tools" and once you get it installed, if you can't figure out how to use it, let me know and I can help.

**Comment by famda** [comment]
This would be much appreciated.
Do you see this being supported?

**Comment by naikrovek** [comment]
no, Creo View is a proprietary format that is itself derived from another proprietary format (Creo Parametric) for which conversion tools already exist.

I reverse engineered the Creo View file format for my own needs (I didn't finish all the parametric stuff, and I finished none of the PVS file format, but all of the mesh-based stuff and material and appearance stuff in the .ol files was done) and then I got a bit of a talking to at work for violating our license agreement, and was forced to destroy all of my notes and code.  That was years ago, and I don't remember any details about any of it anymore, except how far I got, because I really, really wanted to continue.  it was fun.  it's a bit of a shame because I thought the innards of the format were very good.  it made sense, which is something I can say for very few file formats in use today.

Creo View is a niche format in use by few in the overall 3D model ecosystem so I don't see it gaining support anywhere outside of Creo tooling, unfortunately.

I would be happy to do it again, but Creo customers can't ask me to do it (that also goes against the license agreement) and I don't have any model files to look at anyways.  And Creo customers can't give me models to convert, because that constitutes a request to reverse engineer the format given what I've said here.  on top of that, I can't find any publicly available models that exercise a significant fraction of the file format capabilities.  I would need a large number of complex model files to do this correctly.

it just isn't worth it. use an existing converter on your Creo Parametric models and then you can control tessellation and everything else that you would need to control. Creo View conversion tools don't really make sense.

Sorry.

**Comment by naikrovek** [comment]
the ASCII versions of these files may be able to be supported, but Creo View is such a niche format I don't see it happening.

**Comment by famda** [comment]
There are a lot of people asking for these types of conversions (pvz to something else).
Big companies that use PTC tooling, like windchill.
We are paying a lot per year to be able to convert pvz to gltf/fbx for XR.
Might be a niche but there are no options on the opensource ecosystem to support that niche.

**Comment by naikrovek** [comment]
years ago I reverse engineered a large portion of the binary OL file format, which holds the actual model files in a Creo View file, including all of the tessellated stuff, materials, and appearances.  I documented some of the parametric stuff in the file format, and never finished because I didn't need that stuff.

I could convert hundreds of thousands of OL files to glTF 2.0 in 30-45 minutes.  4ms on average for the files I had, and some were several hundred MB in size.

I then got in Big Trouble with my employer, and was forced to destroy my notes, my code, and every model I converted.

I did not reverse the binary PVS file format, so I could not even convert a .pvz assembly.  I was working toward that when I was stopped.

I would be glad to repeat this effort after I retire or no longer work at a company with any agreements with PTC.

A PTC licensee can't ask anyone to reverse the file format (it's against the license agreement to ask) and I can't do it while I work for a PTC licensee.  

to give an indication of difficulty, the OL file format was the very first file type I reverse engineered, and I had no reverse engineering experience prior to that effort.

I wish I could help here, and I can't.

I recommend that PTC licensees convert the binary PVS and OL files to human-readible ASCII and write code to convert that to glTF if you can.

**Comment by naikrovek** [comment]
I haven't worked on this in a long time, but I believe that `pvaroptimize.exe` will get you directly from .pvz to .glb or .gltf.  I don't remember which product included that tool, but I remember that it was not Creo View Files Tools, but another tool that my employer was licensed to use.

sorry that I can't help more.

**Comment by blkot** [comment]
> 

thank you for taking the time, and i really appreciate your advice. just googled it, i believe it's VuforiaStudio, which includes the pvaroptimize.exe tool.

btw, good news is we maximized our pipeline resource on CPUs and Mem for a trial on the big step file, it succeeded, bad news is i don't think we should set this as our default pipeline resource, it's expensive and could also jam our other pipelines.

anyway, your effort is quite important to our investigation progress, thanks again!

**Comment by tellypresence** [comment]
Mentioned in epic #6097, if someone picks this up please reopen

**Comment by tellypresence** [comment]
Not sure where the documentation is for converting Creo View files to ASCII.  

This is from search "AI Overview," not sure if this is even relevant or works:
```
To convert a Creo View file to ASCII, use the command-line tool illustrate2pvbatch provided by PTC, which is 
part of the Creo View Adapters. While there isn't a direct, user-friendly GUI button to export a Creo View file 
as ASCII, you can use this command-line utility to publish a binary file into an ASCII representation within a 
specified output directory. 
```
Looks like a license is required
- [Configuring the Creo View Adapter for Creo Illustrate](https://support.ptc.com/help/creo/view/r9.0/de/index.html#page/creo_view/adapter_install_config_guide/Configuring_Creo_Illustrate_Adapter.html)

**Comment by naikrovek** [comment]
yeah i've learned a lot since i opened this PR.  you were correct to close it, and if i had known more at the time, I would not have opened it.
